backend/app/core/middleware.py

```
@@ -0,0 +1,23 @@
+
+
+async def http_request_logger(request: Request, call_next) -> Response:
+    start_time = time.time()
+    try:
+        response = await call_next(request)
+    except Exception as e:
+        logger.exception("Unhandled exception during request")
+        raise
+
+    process_time = (time.time() - start_time) * 1000  # ms
+    client_ip = request.client.host if request.client else "unknown"
```

Ruff (0.14.11)

Local variable `client_ip` is assigned to but never used

Remove assignment to unused variable `client_ip`

(F841)